poncho/src/poncho/library_network_code.py

```diff
@@ -232,14 +218,12 @@ def start_function(in_pipe_fd, thread_limit=1):
                     os._exit(exit_status)
             elif p < 0:
                 stdout_timed_message(f"TASK {function_id} error: unable to fork to execute {function_name}")
-                return -1
+                return -1, function_id
```

Comment:
Same thing here, this should be the library's fault instead of the function's fault, therefore the library should die rather than tell the worker that the function fails its execution.